README-FIPS.md

```diff
@@ -1,4 +1,445 @@
+the [README-PROVIDERS](README-PROVIDERS.md) file for further details about
+providers.
+
+Installing the FIPS module
+==========================
+
+Once OpenSSL has been built and installed you will need to take explicit steps
+to complete the installation of the FIPS module (if you wish to use it). The
+OpenSSL 3.0 FIPS support is in the form of the FIPS provider which, on Unix, is
+in a `fips.so` file. On Windows this will be called `fips.dll`. Following
+installation of OpenSSL 3.0 the default location for this file is
+`/usr/local/lib/ossl-modules/fips.so` on Unix or
+`C:\Program Files\OpenSSL\lib\ossl-modules\fips.dll` on Windows.
+
+To complete the installation you need to run the `fipsinstall` command line
+application. This does 2 things:
+
+- Runs the FIPS module self tests
+- Generates FIPS module config file output containing information about the
+module such as the self test status, and the module checksum.
+
+The FIPS module must have the self tests run, and the FIPS module config file
+output generated on every machine that it is to be used on. You must not copy
+the FIPS module config file output data from one machine to another.
+
+For example, to install the FIPS module to its default location on Unix:
+
+    $ openssl fipsinstall -out /usr/local/ssl/fipsmodule.cnf -module /usr/local/lib/ossl-modules/fips.so
+
+If you installed OpenSSL to a different location, you need to adjust the output
+and module path accordingly.
+
+
+Using the FIPS Module in applications
+=====================================
+
+There are a number of different ways that OpenSSL can be used in conjunction
+with the FIPS module. Which is the correct approach to use will depend on your
+own specific circumstances and what you are attempting to achieve. Note that the
+old functions `FIPS_mode()` and `FIPS_mode_set()` are no longer present so you
+must remove them from your application if you use them.
+
+Applications written to use the OpenSSL 3.0 FIPS module should not use any
+legacy APIs or features that avoid the FIPS module. Specifically this includes:
+
+- Low level cryptographic APIs (use the high level APIs, such as EVP, instead)
+- Engines
+- Any functions that create or modify custom "METHODS" (for example
+`EVP_MD_meth_new`, `EVP_CIPHER_meth_new`, `EVP_PKEY_meth_new`, `RSA_meth_new`,
+`EC_KEY_METHOD_new`, etc.)
+
+All of the above APIs are deprecated in OpenSSL 3.0 - so a simple rule is to
+avoid using all deprecated functions.
+
+Making all applications use the FIPS module by default
+------------------------------------------------------
+
+One simple approach is to cause all applications that are using OpenSSL to only
+use the FIPS module for cryptographic algorithms by default.
+
+This approach can be done purely via configuration. As long as applications are
+built and linked against OpenSSL 3.0 and do not override the loading of the
+default config file or its settings then they will automatically start using the
```

Comment:
I would replace `will` with `can` as that makes it clear that you need to change the default configuration file according to the following instructions.